syntax = "proto3";

package tensorflow.quantization;

import "tensorflow/core/framework/graph.proto";

option cc_enable_arenas = true;

// Represents an exported TensorFlow model. It consists of a GraphDef and extra
// metadata required for building a SavedModel. This message is primarily used
// to "export" the model produced from various quantization passes in c++ to
// Python layer.
// Next ID: 8
message ExportedModel {
  reserved 4;
  reserved 'variable_shared_names';

  GraphDef graph_def = 1;

  // Name of the initialization node (TF Operation) used for initializing
  // resources like hash tables upon loading.
  string init_node_name = 2;

  // Name of the restore node. When fetched it runs the `RestoreV2` op that
  // restores variables from the checkpoint file specified by `checkpoint_dir`.
  string restore_node_name = 3;

  // Name of the save node. When fetched it runs the `SaveV2` op that saves
  // variables to the checkpoint file specified by `checkpoint_dir`. The op
  // should output the file prefix (equal to the checkpoint_dir) string tensor.
  string save_node_name = 7;

  // Path to the directory where checkpoint files are saved. This directoy is
  // not expected to be persistent (usually a temporary directory). When
  // fetching the restore op (see `restore_node_name`), this value is provided
  // to the "file_prefix" tensor to identify the checkpoint directory.
  string checkpoint_dir = 5;

  // Function name -> function alias mapping. This associates the quantized
  // functions to the original functions' aliases. This information will be used
  // to populate `MetaInfoDef`s `function_aliases` when the quantized model is
  // exported to the saved model. This field is usually only populated for the
  // TF2 models.
  map<string, string> function_aliases = 6;
}
